src/viam/sdk/tests/test_motion.cpp

```diff
@@ -164,11 +165,14 @@ BOOST_AUTO_TEST_SUITE(test_motion_client_server)
 
 BOOST_AUTO_TEST_CASE(test_move_and_get_pose) {
     auto mock = std::make_shared<MockMotion>("mock_motion");
-    client_to_mock_pipeline<Motion>(mock, [](Motion& client) {
+    client_to_mock_pipeline<Motion>(mock, [&](Motion& client) {
         std::string destination_frame("destination");
         std::vector<WorldState::transform> transforms;
         ProtoStruct extra = fake_map();
+        std::string debug_key = "debug-key";
+        extra = add_debug_entry(std::move(extra), debug_key);
```

Comment:
I think now that `add_debug_entry` is void this will need a change, maybe you can change the `get_pose` call below to be 

```cpp
pose_in_frame pose = client.get_pose(fake_component_name(), destination_frame, {}, with_debug_entry(fake_map(), debug_key));
```

and the above two lines deleted, then technically we'll be testing both the `add` and `move` version.

Other than that LGTM though, and feel free to merge without re-requesting review